src/dataflow/split_h5mu/script.py

```diff
@@ -100,6 +101,11 @@ def main():
         mdata_obs.mod[par["modality"]] = adata_obs
         mdata_obs.write_h5mu(output_dir / mdata_obs_name, compression=par["output_compression"])
 
+        # avoid keeping files in memory
+        del mdata_obs
+        del adata_obs
+        gc.collect()
```

Comment:
python should automatically garbage collect when all references to an object are gone. So I think `del` is enough here.

Reply:
I tested this locally and it makes a difference, memory is not immediately reclaimed without the gc.collect() - automatic process doesn't keep up with the loop and memory buildup lingers